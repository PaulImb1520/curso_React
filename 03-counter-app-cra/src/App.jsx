import PropTypes from "prop-types";
const App = ({ title, subTitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};
App.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};
App.defaultProps = {
  name: "Pa√∫l Imbaquingo",
  subTitle: "Ejecutor de Justicia de Neuvillete",
  //title: "Primer sabio de Buer",
};
export default App;
