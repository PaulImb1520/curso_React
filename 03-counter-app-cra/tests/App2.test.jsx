/** @jest-environment jsdom*/
import { render, screen } from "@testing-library/react";
import App from "../src/App";

describe("Pruebas en <App/>", () => {
  const title = "Soy un título";
  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<App title={title} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar el mensaje 'Soy un título'", () => {
    render(<App title={title} />);
    expect(screen.getByText(title)).toBeTruthy;
  });

  test("Debe mostrar el título en un h1", () => {
    render(<App title={title} />);
    expect(screen.getByText(title)).toBeTruthy;
  });
});
