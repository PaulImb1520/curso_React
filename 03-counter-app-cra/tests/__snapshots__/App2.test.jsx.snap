// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <App/> Debe haacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Soy un título
  </h1>
  <p>
    Ejecutor de Justicia de Neuvillete
  </p>
  <p>
    Paúl Imbaquingo
  </p>
</div>
`;

exports[`Pruebas en <App/> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Soy un título
  </h1>
  <p>
    Ejecutor de Justicia de Neuvillete
  </p>
  <p>
    Paúl Imbaquingo
  </p>
</div>
`;
